src/storage/SparqlResourceStore.ts

```diff
@@ -0,0 +1,753 @@
+                quad(variable('s'), variable('p'), variable('o')),
+              ],
+            },
+          ],
+        },
+      ],
+      type: 'update',
+      prefixes: {},
+    } as Update;
+    return this.sendSparqlUpdate(deleteContainerQuery);
+  }
+
+  /**
+   * Checks whether the specified container is empty.
+   * Ignores the .metadata file corresponding to the container.
+   * @param containerURI - Identifier of the container.
+   */
+  private async isEmptyContainer(containerURI: string): Promise<boolean> {
+    const containerQuery = {
+      queryType: 'ASK',
+      where: [
+        {
+          type: 'graph',
+          name: namedNode(containerURI),
+          triples: [
+            quad(namedNode(containerURI), CONTAINS_PREDICATE, variable('o')),
+            {
+              type: 'filter',
+              expression: {
+                type: 'operation',
+                operator: '!=',
+                args: [
+                  variable('o'),
+                  namedNode(`${containerURI}.metadata`),
+                ],
+              },
+            },
+          ],
+        },
+      ],
+      type: 'query',
+    } as unknown as AskQuery;
+    const result = await this.sendSparqlQuery(containerQuery);
+    return !result.boolean;
+  }
+
+  /**
+   * Helper function without extra validation to get all triples in a graph corresponding to the specified URI.
+   * @param URI - URI of the resource.
+   */
+  private async getSparqlRepresentation(URI: string): Promise<any> {
+    const representationQuery = {
+      queryType: 'CONSTRUCT',
+      template: [
+        quad(variable('s'), variable('p'), variable('o')),
+      ],
+      where: [
+        {
+          type: 'graph',
+          name: namedNode(URI),
+          patterns: [
+            {
+              type: 'bgp',
+              triples: [
+                quad(variable('s'), variable('p'), variable('o')),
+              ],
+            },
+          ],
+        } as GraphPattern,
+      ],
+      type: 'query',
+      prefixes: {},
+    } as ConstructQuery;
+    return (await this.sendSparqlQuery(representationQuery)).results.bindings;
+  }
+
+  /**
+   * Helper function to get the representation of a document resource.
+   * @param resourceURI - Identifier of the resource to retrieve.
+   */
+  private async getResourceRepresentation(resourceURI: string): Promise<Representation> {
+    // Get the triples from the resourceURI graph and from the corresponding metadata graph.
+    const data: Quad[] = await this.getSparqlRepresentation(resourceURI);
+    const metadata: Quad[] = await this.getSparqlRepresentation(`${resourceURI}.metadata`);
+
+    // Only include the triples of the resource graph in the data readable.
+    const readableData = streamifyArray([ ...data ]);
+    return {
+      dataType: DATA_TYPE_QUAD,
+      data: readableData,
+      metadata: {
+        raw: metadata,
+        contentType: CONTENT_TYPE_QUADS,
+      },
+    };
```

Comment:
Could have a function that takes as input a readable and set of quads and returns this.